mock/py/mockbuild/config.py

```diff
@@ -398,6 +404,53 @@ def multiply_platform_multiplier(config_opts):
     if '%_platform_multiplier' not in config_opts["macros"]:
         config_opts["macros"]["%_platform_multiplier"] = 10 if config_opts["forcearch"] else 1
 
+
+def process_isolated_build_config(cmdline_opts, config_opts):
+    """
+    Read the json file generated by the previous --calculate-build-dependencies
+    run, and adjust the current set of options in CONFIG_OPTS.
+    """
+
+    if not cmdline_opts.isolated_build_config:
+        return
+
+    config_opts["isolated_build"] = True
+
+    json_conf, repo_reference = cmdline_opts.isolated_build_config
+    with open(json_conf, "r", encoding="utf-8") as fd:
+        data = json.load(fd)
+
+    if not data["config"].get("bootstrap_image_ready"):
+        raise exception.BadCmdline(
+            f"The file {json_conf} did not record the bootstrap_image_ready=True "
+            "config which means we are not able to prepare the bootstrap chroot "
+            "in an isolated mode.")
+
+    for opt, val in data["config"].items():
+        config_opts[opt] = val
```

Comment:
I would rather utilize https://github.com/rpm-software-management/mock/blob/main/mock/py/mockbuild/config.py#L679 It will need some alternation, but you get all these includes for free.

Reply:
The update_config_from_file() works with filenames though, and does a lot of magic with includes... while here we don't want to interpret includes, and we don't even have a "Python file / Mock config file" in hand; we have just a poor dictionary object in hand that we want to merge with config_opts...   I created a new utility method `update_config_from_dict` which logically complements `update_config_from_file` so this fact is clear.... though.... I may be missing something.

The rest of this method should otherwise stay as-is IMO, this seems to be the correct place to set "isolated build" defaults (e.g. set the `bootstrap_image_skip_pull = True`).